demos/privacy_preservation/pose.py

```diff
@@ -0,0 +1,93 @@
+import concurrent.futures
+from threading import Thread, current_thread
+import time
+import numpy as np
+import tensorflow as tf
+
+
+class Pose:
+    """
+    Class that continuously detects objects on image using a dedicated thread.
+    """
+
+    def __init__(self, path, num_threads, det_pose_queue, pose_postprocessor_queue, frames):
+        if "lightning" in path:
+            self.shape = 192
+        else:
+            self.shape = 256
+        self.frame = None # Full frame
+        self.humans = None
+        self.people = None
+        self.stopped = False
+        self.det_pose_queue = det_pose_queue
+        self.pose_postprocessor_queue = pose_postprocessor_queue
+        self.frames = frames
+        self.interpreters = {}
+        self.executor = concurrent.futures.ThreadPoolExecutor(max_workers=10, initializer=self.init_threads, initargs=(self.interpreters, path, num_threads))
+    
+    def init_threads(self, interpreters, model_path, num_threads):
+        model = tf.compat.v1.lite.Interpreter(model_path=model_path, num_threads=num_threads)
+        model.allocate_tensors()
+        interpreters[current_thread().name] = model
+
+    def run(self, interpreters, image, idx):
+        model = interpreters[current_thread().name]
+        model.set_tensor(model.get_input_details()[0]['index'], image)
+        st = time.time()
+        model.invoke()
+        end = time.time()
+        # print("pose", end - st)
+        output_tensor = model.get_tensor(model.get_output_details()[0]['index'])
+        return output_tensor, idx
+
+    def start(self):
+        self.stopped = False
+        Thread(target=self.detect, args=()).start()
+        return self
+
+    def detect(self):
+        while not self.stopped:
+            idx = self.det_pose_queue.get()
+            if idx is None:
+                for i in range(8):
```

Comment:
Worth keeping values like 8 here in some config file or as a global variable - named and perhaps explained.